### PR TITLE
BSTOR-13 - i18n

### PR DESCRIPTION
**Pull request by Pacane**
i18n server + basic SEO
Did a bit of translations
Added support for i18n in Nametokens for uiBinder as well

### CODE REVIEW

File: client/src/main/resources/com/arcbees/beeshop/client/resources/AppMessages_fr.properties [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#
```
Comment (Pacane) [inline client/src/main/resources/com/arcbees/beeshop/client/resources/AppMessages_fr.properties]: @jclariviere should there be a license header in .properties?

File: client/pom.xml [hunk 2]
```diff
@@ -76,6 +76,14 @@
             <groupId>org.vectomatic</groupId>
             <artifactId>lib-gwt-svg</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
```
Comment (olafleur) [inline client/pom.xml:86]: no version number?

File: client/src/main/java/com/arcbees/beeshop/client/application/ApplicationView.java [hunk 3]
```diff
@@ -16,15 +16,14 @@
 
 package com.arcbees.beeshop.client.application;
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/application/ApplicationView.java:17]: Is it beeshop of beestore?

File: client/src/main/java/com/arcbees/beeshop/client/application/gin/BootstrapperImpl.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package com.arcbees.beeshop.client.application.gin;
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/application/gin/BootstrapperImpl.java:14]: Copyright notice

File: client/src/main/java/com/arcbees/beeshop/client/application/gin/BootstrapperImpl.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import com.google.gwt.user.client.History;
+import com.gwtplatform.mvp.client.Bootstrapper;
+import com.gwtplatform.mvp.client.proxy.PlaceManager;
+import com.gwtplatform.mvp.shared.proxy.PlaceRequest;
+
+public class BootstrapperImpl implements Bootstrapper {
+    private final PlaceManager placeManager;
+
+    @Inject
+    public BootstrapperImpl(
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/application/gin/BootstrapperImpl.java]: rm public

File: client/src/main/java/com/arcbees/beeshop/client/application/gin/ClientModule.java [hunk 6]
```diff
@@ -16,10 +16,11 @@
 
 package com.arcbees.beeshop.client.application.gin;
 
-import com.arcbees.beeshop.client.NameTokens;
+import com.arcbees.beeshop.common.NameTokens;
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/application/gin/ClientModule.java:19]: common or shared?

File: client/src/main/java/com/arcbees/beeshop/client/application/ApplicationView.java (hunk 3 above)
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/ApplicationView.java:17]: it's beestore, however, all the packages use that name. We planned to rename it, but it's still not done. It'll have to be done in a separate PR though.

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml [hunk 7]
```diff
@@ -11,7 +14,9 @@
         <g:HTMLPanel tag="section" ui:field="mainProducts" styleName="{page.style.mainProducts}"/>
 
         <section class="{res.style.stripe} {res.style.txt_center} {page.style.shoppingByProduct}">
-            <h3>shopping by products</h3>
+            <h3>
+                <ui:msg description="shopping by projects">shopping by products</ui:msg>
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml]: projects or products?

File: client/pom.xml (hunk 2 above)
Comment (Pacane) [inline client/pom.xml:86]: the version number is specified in the parent pom in the dependency management section.

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml [hunk 8]
```diff
@@ -97,9 +102,13 @@
         </section>
 
         <section class="{res.style.stripe} {page.style.welcome}">
-            <h2>Welcome to the Beestore</h2>
-            <p>The Beestore is a study case from Arcbees
-                <span>to show our hive of products and branded gears. Enjoy!</span>
+            <p>
+                <ui:msg description="case study">The Beestore is a case study from Arcbees</ui:msg>
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml:106]: I've already seen that sentence earlier...

File: client/src/main/java/com/arcbees/beeshop/client/resources/NameTokensConstants.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package com.arcbees.beeshop.client.resources;
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/resources/NameTokensConstants.java:17]: copyright

File: common/pom.xml [hunk 10]
```diff
@@ -16,6 +16,10 @@
             <groupId>org.jboss.resteasy</groupId>
             <artifactId>resteasy-jaxrs</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
```
Comment (olafleur) [inline common/pom.xml]: version number?

File: client/src/test/java/com/arcbees/beeshop/client/application/ApplicationPresenterTest.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
+package com.arcbees.beeshop.client.application;
```
Comment (olafleur) [inline client/src/test/java/com/arcbees/beeshop/client/application/ApplicationPresenterTest.java:14]: copyright

File: server/src/main/java/com/arcbees/beeshop/server/servlets/NotFoundServlet.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
+package com.arcbees.beeshop.server.servlets;
```
Comment (olafleur) [inline server/src/main/java/com/arcbees/beeshop/server/servlets/NotFoundServlet.java:14]: copyright

File: server/src/main/java/com/arcbees/beeshop/server/servlets/RootServlet.java [hunk 13]
```diff
@@ -2,7 +2,9 @@
```
Comment (olafleur) [inline server/src/main/java/com/arcbees/beeshop/server/servlets/RootServlet.java:15]: copyright

File: client/pom.xml (hunk 2 above)
Comment (olafleur) [inline client/pom.xml:86]: right.

File: client/src/main/java/com/arcbees/beeshop/client/application/ApplicationView.java (hunk 3 above)
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/application/ApplicationView.java:17]: ok

**Comment by olafleur** [comment]
LGTM

File: client/src/main/java/com/arcbees/beeshop/client/application/gin/ClientModule.java (hunk 6 above)
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/gin/ClientModule.java:19]: It was already called common.

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml (hunk 8 above)
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml:106]: Yeah, I can't "not repeat it" sadly. It's there in the page and in the footer.

**Comment by Pacane** [comment]
@olafleur thanks for the review! :)

**Comment by olafleur** [comment]
No problem! ;)

**Comment by jclariviere** [comment]
LGTM, but you might want @meriouma and/or @Chris-V to look at it since they did the i18n on the website.

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml (hunk 8 above)
Comment (meriouma) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml:106]: Maybe you could move those to a Constants interface. I believe you could use them with an `ui:with` field

**Comment by meriouma** [comment]
LGTM

File: client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml (hunk 8 above)
Comment (Pacane) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml:106]: talking about the description or the text in the element?

Comment (meriouma) [inline client/src/main/java/com/arcbees/beeshop/client/application/home/HomeView.ui.xml:106]: The text, but you would have to remove the `ui:msg` tag

**Comment by Chris-V** [comment]
LGTM
